starport/templates/typed/new_stargate.go

```diff
@@ -276,21 +303,20 @@ func (t *typedStargate) clientRestRestModify(opts *Options) genny.RunFn {
 		content := strings.Replace(f.String(), placeholder2, replacement, 1)
 
 		template = `%[1]v
-    r.HandleFunc("/custom/%[2]v/" + types.QueryList%[3]v, list%[3]vHandler(clientCtx)).Methods("GET")
+    r.HandleFunc("/%[2]v/%[3]v/{id}", get%[4]vHandler(clientCtx)).Methods("GET")
+    r.HandleFunc("/%[2]v/%[3]v", list%[4]vHandler(clientCtx)).Methods("GET")
 `
-		replacement = fmt.Sprintf(template, placeholder3, opts.ModuleName, strings.Title(opts.TypeName))
+		replacement = fmt.Sprintf(template, placeholder3, opts.ModuleName, pluralize.NewClient().Plural(opts.TypeName), strings.Title(opts.TypeName))
 		content = strings.Replace(content, placeholder3, replacement, 1)
 
-		template = `%s
-    r.HandleFunc("/custom/%s/%s", create%sHandler(clientCtx)).Methods("POST")
+		template = `%[1]v
+    r.HandleFunc("/%[2]v/%[3]v", create%[4]vHandler(clientCtx)).Methods("POST")
+    r.HandleFunc("/%[2]v/%[3]v/{id}", set%[4]vHandler(clientCtx)).Methods("PUT")
```

Comment:
We should use `POST` for every message. Because, whatever the transaction does, at the http level it's just about sending a message to the Blockchain it should be agnostic of the actual action of the message

Reply:
solved by 4d69a84